src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -139,10 +145,43 @@ private void InitializeCore()
         }
 
         private void UpdateStateMachine()
+        {
+            if (!TryUpdateStateMachine())
+            {
+                Task.Factory.SafeStartNew(LoopOnBackgroundAsync, CancellationToken.None, TaskScheduler.Default);
+            }
+        }
+
+        private async Task LoopOnBackgroundAsync()
+        {
+            for (var i = 0; i < 10; i++)
```

Comment:
okay, rather than InitializeAsync, this seems better place to wait for solution to be fully loaded.